core/connection/manager.go

```diff
@@ -507,14 +552,14 @@ func (m *connectionManager) createP2PSession(ctx context.Context, c Connection,
 		defer log.Trace().Msg("Cleaning: requesting session destroy DONE")
 
 		sessionDestroy := &pb.SessionInfo{
-			ConsumerID: consumerID.Address,
+			ConsumerID: opts.ConsumerID.Address,
 			SessionID:  sessionResponse.GetID(),
 		}
 
 		log.Debug().Msgf("Sending P2P message to %q: %s", p2p.TopicSessionDestroy, sessionDestroy.String())
 		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 		defer cancel()
-		_, err := p2pChannel.Send(ctx, p2p.TopicSessionDestroy, p2p.ProtoMessage(sessionDestroy))
+		_, err := m.channel.Send(ctx, p2p.TopicSessionDestroy, p2p.ProtoMessage(sessionDestroy))
```

Comment:
I think it would be nice to name this channel in an explicit way even if it's longer. Channels could be many things, so it would provide more context.

Reply:
Good point about naming, but this will affect more changes, let's make it a separate PR.